### PR TITLE
feat: truncate text with button for popover element GH-11475

### LINKED ISSUE #11475: Component needed which will truncate text and show full text in popover
There is needed component which will truncate text, show more button and on click will open popover with full text. This feature is required to saved carts epic for saved carts list.

### PR DESCRIPTION
**Pull request by dydome**
close #11475

### CODE REVIEW

File: projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+   */
+  transform(value: string, args?: any[]): string {
+    if (!args) {
+      return value;
+    }
+
+    const limit =
+      args.length > 0 &&
+      args[0] !== null &&
+      args[0] !== undefined &&
```
Comment (mszczepaniak) [inline projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate.pipe.ts]: we could write `args.length > 0 && ars[0] && Number.isInteger(+args[0])` instead

File: projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      Number.isInteger(+args[0])
+        ? parseInt(args[0], 10)
+        : 20;
+
+    let trail = '...';
```
Comment (mszczepaniak) [inline projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate.pipe.ts]: let's move that to the top of the method

File: projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    if (
+      args.length > 1 &&
+      args[1] !== null &&
+      args[1] !== undefined &&
```
Comment (mszczepaniak) [inline projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate.pipe.ts]: same as with args[0] comment

File: projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate-text-with-popover.module.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { I18nModule } from '@spartacus/core';
+import { PopoverModule } from '../popover/popover.module';
+import { TruncateTextWithPopoverComponent } from './truncate-text-with-popover.component';
+import { TruncatePipe } from './truncate.pipe';
+
+@NgModule({
+  imports: [CommonModule, I18nModule, PopoverModule],
```
Comment (mszczepaniak) [inline projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate-text-with-popover.module.ts:9]: I don't think that you need `I18nModule` here

File: projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate-text-with-popover.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
+import { I18nTestingModule } from '@spartacus/core';
+import { KeyboardFocusTestingModule } from 'projects/storefrontlib/src/layout/a11y/keyboard-focus/focus-testing.module';
+import { TruncateTextWithPopoverComponent } from './truncate-text-with-popover.component';
+import { TruncateTextWithPopoverModule } from './truncate-text-with-popover.module';
+
+const mockContent = 'Test text';
+
+fdescribe('TruncateTextWithPopoverComponent', () => {
```
Comment (mszczepaniak) [inline projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate-text-with-popover.component.spec.ts:9]: remove fdescribe

File: projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate-text-with-popover.module.ts (hunk 4 above)
Comment (dydome) [reply projects/storefrontlib/src/shared/components/truncate-text-with-popover/truncate-text-with-popover.module.ts:9]: button content `{{ 'common.more' | cxTranslate }}`

**Pull request closed without merging**